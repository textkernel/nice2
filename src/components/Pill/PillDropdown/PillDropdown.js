import * as React from 'react';
import PropTypes from 'prop-types';
import { bem } from '../../../utils/bem';
import { Button } from '../../Buttons';
import styles from './PillDropdown.scss';

const { block, elem } = bem('PillDropdown', styles);

export const PillDropdown = React.forwardRef((props, ref) => {
    const { close, doneLabel, noPadding, children, ...rest } = props;

    const handleDoneClick = () => {
        close();
    };

    return (
        <div ref={ref} role="presentation" {...rest} {...block(props)}>
            <div role="dialog" {...elem('dialog', props)}>
                <div {...elem('content', props)}>{children({ close })}</div>
                <div {...elem('footer', props)}>
                    <Button context="primary" size="small" onClick={handleDoneClick}>
                        {doneLabel}
                    </Button>
                </div>
            </div>
        </div>
    );
});

PillDropdown.displayName = 'PillDropdown';

PillDropdown.propTypes = {
    /** The dropdown content renderer function. It is called with:
     *   * close {function} - see below
     */
    children: PropTypes.func.isRequired,
    /** function that closes the dropdown */
    close: PropTypes.func,
    /** label for the Done button */
    doneLabel: PropTypes.string.isRequired,
    /** whether or not to add a padding to the dropdown container.
     * It is useful if you need to add elements that stretch to the edge of the container.
     * You can use the CSS variable --pill-dropdown-padding to add padding as required.
     * */
    noPadding: PropTypes.bool,
};

PillDropdown.defaultProps = {
    close: null,
    noPadding: false,
};
