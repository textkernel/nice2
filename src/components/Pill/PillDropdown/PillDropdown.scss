:root {
    --pill-dropdown-padding: var(--spacing-3x);
}

@keyframes dropdownAppear {
    from {
        opacity: 0;
        transform: translateY(-15px) scale(0.9);
    }
    to {
        opacity: 1;
        transform: translateY(0) scale(1);
    }
}

.PillDropdown {
    background: var(--color-background);
    margin: var(--spacing-2x) 0;
    border-radius: var(--border-radius);
    border: 1px solid var(--color-neutral);
    box-shadow: 0 5px 10px rgba(0, 0, 0, 0.1);
    transform-origin: top left;
    animation: dropdownAppear var(--transition-duration-quick) forwards;
    animation-timing-function: linear;

    &__content {
        padding: var(--pill-dropdown-padding);

        &--noPadding {
            padding: 0;
        }
    }

    &__footer {
        display: flex;
        border-top: solid 1px var(--color-neutral-40);
        justify-content: flex-end;
        padding: var(--pill-dropdown-padding);
    }
}
