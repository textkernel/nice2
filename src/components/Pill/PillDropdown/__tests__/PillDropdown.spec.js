import React from 'react';
import toJson from 'enzyme-to-json';
import PillDropdown from '../PillDropdown';

describe('<PillDropdown> component', () => {
    const childrenMock = jest.fn();
    const closeMock = jest.fn();

    let wrapper;

    beforeEach(() => {
        wrapper = mount(<PillDropdown close={closeMock}>{childrenMock}</PillDropdown>);
    });

    afterEach(() => {
        jest.resetAllMocks();
    });

    it('should render correctly', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    it('should render correctly without padding', () => {
        wrapper.setProps({ noPadding: true });
        wrapper.update();

        expect(wrapper.find('.PillDropdown--noPadding')).toHaveLength(1);
    });
    it('should call children function with close as arguments', () => {
        expect(childrenMock).toHaveBeenCalledWith({
            close: closeMock,
        });
    });
});
