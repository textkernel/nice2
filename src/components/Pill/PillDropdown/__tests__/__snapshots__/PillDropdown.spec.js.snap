// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PillDropdown> component should render correctly 1`] = `
<PillDropdown
  close={[MockFunction]}
  noPadding={false}
>
  <div
    role="presentation"
  >
    <div
      className="PillDropdown"
      role="dialog"
    />
  </div>
</PillDropdown>
`;
